### DISCUSSION TITLE
tutorials/scripting/scene_unique_nodes

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#267] [User-contributed notes]
# tutorials/scripting/scene_unique_nodes

Introduction: Using get_node() to reference nodes from a script can sometimes be fragile. If you move a button in a UI scene from one panel to another, the button's node path changes, and if a scri...

https://docs.godotengine.org/en/stable/tutorials/scripting/scene_unique_nodes.html

**Comment by nanodeath** [comment]
It's interesting that the same-scene behavior is described as a _limitation_ when actually it's a _feature_. If it wasn't for that functionality, unique nodes in instantiated scenes would be unusable because they wouldn't necessarily be unique, if the scene that instantiated them instantiated multiple nodes. That is, it's a way of guaranteeing uniqueness in a way that makes it a useful feature for _all_ scenes, not just the main one.

In the example above, what would happen if we had two Swords, and each one had a `%Hilt` node? Would be problematic if Player could say `GetNode("%Hilt")`.

**Comment by Synzorasize** [comment]
I think it should be mentioned somewhere that root nodes cannot be scene-unique nodes in their own scene. Only if the scene is instantiated in another scene, then it can be a scene-unique node in _that_ scene, but not its own. Which makes sense, because node outside the scene of the node cannot refer to that unique node and you don't need a unique node to refer to itself, since you can just use `self`!
